src/app/services/baw-api/base-api.service.ts

```diff
@@ -62,126 +64,72 @@ export abstract class BawApiService {
   }
 
   /**
-   * Get response from details route
-   * @param next Callback function for successful response
-   * @param error Callback function for failed response
-   * @param path API path
-   * @param args URL arguments
-   * @param filters API filters
+   * Create a http request
+   * @param method Method to use
+   * @param path API Path
+   * @param body Optional body content
    */
-  protected apiList(
-    next: (data: any) => void,
-    error: (err: any) => void,
-    path: string,
-    filters?: Filters
-  ) {
-    if (!filters) {
-      this.httpGet<APIResponse>(path).subscribe(
-        this.handleResponse(next, error)
-      );
-    } else {
-      this.httpPost<APIResponse>(path + "/filter", filters).subscribe(
-        this.handleResponse(next, error)
-      );
-    }
-  }
-
-  /**
-   * Create request for API route
-   * @param next Callback function for successful response
-   * @param error Callback function for failed response
-   * @param path API path
-   * @param body Request body
-   */
-  protected apiCreate(
-    next: (data: any) => void,
-    error: (err: any) => void,
-    path: string,
-    body?: any
-  ) {
-    this.httpPost<APIResponse>(path, body).subscribe(
-      this.handleResponse(next, error)
-    );
-  }
-
-  /**
-   * Update request for API route
-   * @param next Callback function for successful response
-   * @param error Callback function for failed response
-   * @param path API path
-   * @param body Request body
-   */
-  protected apiUpdate(
+  protected apiRequest(
+    method: "LIST" | "FILTER" | "SHOW" | "CREATE" | "UPDATE" | "DELETE",
     next: (data: any) => void,
     error: (err: any) => void,
     path: string,
     body?: object
   ) {
-    this.httpPatch<APIResponse>(path, body).subscribe(
-      this.handleResponse(next, error)
-    );
+    path = this.url + path;
+
+    switch (method) {
+      case "LIST":
+      case "SHOW":
+        return this.http
+          .get<APIResponse>(path)
+          .subscribe(this.handleResponse(next, error));
+      case "FILTER":
+        return this.http
+          .post<APIResponse>(path + "/filter", body)
+          .subscribe(this.handleResponse(next, error));
+      case "CREATE":
+        return this.http
+          .post<APIResponse>(path, body)
+          .subscribe(this.handleResponse(next, error));
+      case "UPDATE":
+        return this.http
+          .patch<APIResponse>(path, body)
+          .subscribe(this.handleResponse(next, error));
+      case "DELETE":
+        return this.http
+          .delete<APIResponse>(path)
+          .subscribe(this.handleResponse(next, error, true));
+    }
```

Comment:
revert this change, please. 

Why not use methods?
Stringly type invocation is not fantastic  - especially since you didn't handle any other case in the switch. (I do soee the string-type restriction in the parameter type, but types get erased at compile time).

Reply:
My bad, misread https://docs.angularjs.org/api/ngResource/service/$resource and thought it was building a single function which handles all the requests instead of a class.